client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/pivot/TermsGroupSource.java

```diff
@@ -42,6 +42,8 @@
         PARSER.declareString(optionalConstructorArg(), FIELD);
         Script.declareScript(PARSER, optionalConstructorArg(), SCRIPT);
         PARSER.declareBoolean(optionalConstructorArg(), MISSING_BUCKET);
+        // either a script or a field must be declared, or both
+        PARSER.declareRequiredFieldSet(FIELD.getPreferredName(), SCRIPT.getPreferredName());
```

Comment:
The parser is for receiving data from the server. Consequently, it should follow the same leniency that the lenient server parser has. 

So, I don't think this declaration should be here (same for the rest of the HLRC group sources).

Reply:
:+1: yes, you are right. I also need to move the other check into the builders, as the constructor is used from the parser.

Reply:
however, I can't find any other place that validates. I will simply revert all HLRC changes.